### PR TITLE
Talk pages - scroll to comment after posting

### PR DESCRIPTION
**Pull request by tonisevener**
**Phabricator:** 
N/A

### Notes
This PR scrolls to a particular comment view after a user successfully posts a new reply. This is based off of #4331, so please do not merge until that one is merged.

### Test Steps
1. Post a reply on a topic. Try the blue Reply button when it's near the bottom of the scroll view - it should append a reply off screen.
2. Confirm that after a brief delay after posting, the new comment scrolls into view.

### CODE REVIEW

File: Wikipedia/Code/TalkPageViewController.swift [hunk 1]
```diff
@@ -303,6 +298,91 @@ class TalkPageViewController: ViewController {
...
+        // MAYBETODO: Sort by date here?
+        
+        guard let newComment = newComments.first else {
+            return nil
+        }
+        
+        return newComment
+    }
+    
+    private func scrollToComment(commentViewModel: TalkPageCellCommentViewModel) {
```
Comment (tonisevener) [inline Wikipedia/Code/TalkPageViewController.swift:345]: My hope is that this is will be reusable-enough for navigating through our eventual find in page results.

File: Wikipedia/Code/TalkPageCell.swift [hunk 2]
```diff
@@ -109,6 +109,20 @@ final class TalkPageCell: UICollectionViewCell {
 
         stackView.addArrangedSubview(topicView)
     }
+    
+    // MARK: - Public
+    
+    func displayingCommentViewForViewModel(_ commentViewModel: TalkPageCellCommentViewModel) -> TalkPageCellCommentView? {
+        
+        return stackView.arrangedSubviews.first { view in
```
Comment (staykids) [inline Wikipedia/Code/TalkPageCell.swift:117]: I think this entire expression can be simplified and read a little clearer as:
```swift
return stackView.arrangedSubviews
            .compactMap { $0 as? TalkPageCellCommentView }
            .first(where: { $0.viewModel == commentViewModel })
```

File: Wikipedia/Code/TalkPageViewController.swift (hunk 1 above)
Comment (staykids) [reply Wikipedia/Code/TalkPageViewController.swift:345]: 👍🏽 I've been playing with a few ideas to simplify this, but so far your solution seems to be working quite nicely for me. I anticipate with find in page we'll probably have to parameterize out the animation properties.

File: Wikipedia/Code/TalkPageViewController.swift [hunk 3]
```diff
@@ -303,6 +298,91 @@ class TalkPageViewController: ViewController {
             WMFAlertManager.sharedInstance.showBottomAlertWithMessage(title, subtitle: nil, image: image, dismissPreviousAlerts: true)
         }
     }
+    
+    // MARK: - Scrolling Helpers
+    
+    private func scrollToLastTopic() {
+        if viewModel.topics.count > 0 {
+            let indexPath = IndexPath.init(item: viewModel.topics.count - 1, section: 0)
```
Comment (staykids) [inline Wikipedia/Code/TalkPageViewController.swift:306]: The `.init` here can be removed.

**Review by staykids — APPROVED** [review]
Nice! Included are some minor optional notes. Happy to merge as-is if you disagree.

File: Wikipedia/Code/TalkPageCell.swift (hunk 2 above)
Comment (tonisevener) [reply Wikipedia/Code/TalkPageCell.swift]: Good call, this is done!

**Pull request merged**